apps/supernova/src/lib/createFiltersSlice.jsx

```diff
@@ -28,11 +28,21 @@ const parsePredefinedFilters = (predefinedFilters) => {
 const parseInitialFilters = (initialFilters, filterLabels) => {
   if (!initialFilters) return initialFiltersState.initialFilters
 
-  if (typeof initialFilters !== "object") {
+  if (typeof initialFilters !== "object" || initialFilters === null) {
     console.warn("[supernova]::parseInitialFilters: initialFilters object is not an object")
     return {}
   }
 
+  // Check if all values are arrays
+  initialFilters = Object.entries(initialFilters).reduce((acc, [key, value]) => {
```

Comment:
**(Non-blocking suggestion)**

Good solution, wonder if this would be more readable instead of manually constructing key-value pairs. But please merge as you want as we need to do this fast, no problem.

```suggestion
  initialFilters = Object.fromEntries(
    Object.entries(initialFilters).filter(([key, value]) => {
      const isArray = Array.isArray(value)
      if (!isArray) {
        console.warn(`[supernova]::parseInitialFilters: Value for "${key}" is not an Array.`)
      }
      return isArray
    })
  )
```